types/helpers/helpers.core.d.ts

```diff
@@ -0,0 +1,136 @@
+/**
+ * An empty function that can be used, for example, for optional callback.
+ */
+export function noop(): any;
+
+/**
+ * Returns a unique id, sequentially generated from a global variable.
+ * @returns {number}
+ * @function
+ */
+export function uid(): number;
+/**
+ * Returns true if `value` is neither null nor undefined, else returns false.
+ * @param {*} value - The value to test.
+ * @returns {boolean}
+ * @since 2.7.0
+ */
+export function isNullOrUndef(value?: any): value is undefined;
+/**
+ * Returns true if `value` is an array (including typed arrays), else returns false.
+ * @param {*} value - The value to test.
+ * @returns {boolean}
+ * @function
+ */
+export function isArray<T = any>(value: any): value is ArrayLike<T>;
+/**
+ * Returns true if `value` is an object (excluding null), else returns false.
+ * @param {*} value - The value to test.
+ * @returns {boolean}
+ * @since 2.7.0
+ */
+export function isObject(value: any): value is object;
+/**
+ * Returns true if `value` is a finite number, else returns false
+ * @param {*} value  - The value to test.
+ * @returns {boolean}
+ */
+export function isFinite(value: any): value is number;
```

Comment:
I see your intent here, but I'm not sure that's a good declaration. Take the following example:

```typescript
function doSomething(x: number | null) {
  if (helpers.isFinite(x)) {
    // x is `number` here
  } else {
    // x is wrongly assumed to be `null` here, when it could be NaN, +Infinity, or -Infinity
  }
}
```

Unfortunately, I don't think TypeScript is capable of describing "x might still be a number if this function returns false", so this function might have to simply return `boolean` for now.

Reply:
but isn't the positive case: 
```js
if (helpers.isFinite(x)) {
    // x is `number` here
  } 
```
more important than the else case?.

Reply:
Not a typescript user, but as the checking is compile time and you can override it, I think the current typing would satisfy the obvious use cases of the function. Its not perfect though, but neither is typescript.

Reply:
Thinking about it, we can work around it using branded types:

```typescript
export function isFinite(value: any): value is FiniteNumber;
type FiniteNumber = number & FiniteNumberBrand;
class FiniteNumberBrand { private __finite__: never; }
```

(`FiniteNumber` and `FiniteNumberBrand` intentionally not exported)

When the function returns `true`, `value` is promoted to `FiniteNumber`, which is effectively the same thing as a normal number; otherwise, it's recognized as still potentially some other number. It's quite awkward (could improve in a future version of TypeScript), but gets the job done.